client/src/components/PostEntry.js

```diff
@@ -0,0 +1,122 @@
+import React, {useState, useEffect} from 'react';
+
+function PostEntry(props) {
+  const [title, setTitle] = useState(props.title);
+  const [image, setImage] = useState(props.image);
+  const [imageSrc, setImageSrc] = useState(props.imageLink);
+  const [imageTitle, setImageTitle] = useState(props.imageLink);
+  const [link, setLink] = useState(props.link);
+  const [subject, setSubject] = useState(props.subject);
```

Comment:
Given the way you have this setup, I would just use the props directly rather than managing a separate internal state. Also, you can use default values for your props rather than passing a bunch of empties:

```suggestion
function PostEntry({
  submitAction,
  title: '',
  imageLink: '',
  link: '',
  subject: '',
  edit: false,
}) {
```

Also, as per my other PR review, I would just send the entire `post` state. You can send its setter as a prop as well, if you need to update it in this component, rather than the large `useEffect` you have here currently. I'm also not fully convinced that this is a better solution than just making this one component with all the logic contained within it, using the `edit` prop to decide what functionality to have, as per my other review. I would review this code with the above in mind and see what makes the most sense to you.